src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointComplications.java

```diff
@@ -0,0 +1,503 @@
+            }
+            return new String(insertedSequenceBytes);
+        } else {
+            return "";
+        }
+    }
+
+    // TODO: 03/03/17 this complicated tandem duplication case is not exactly reproducible (e.g. '+' and '-' strand may give slightly different results by this treatment)
+    private static final class TandemRepeatComplication {
+        final int lowerRepeatNumberEstimate;
+        final int higherRepeatNumberEstimate;
+        final int repeatedSeqLen;
+        final int pseudoHomologyLen;
+
+        /**
+         * This function, when given overlaps of two corresponding regions on reference and contig sequences,
+         * attempts to find--naively and slowly--the repeat numbers on the reference and on the contig of tandem repeats,
+         * as well as the pseudo-homology between the duplicated sequence and the right flanking region.
+         *
+         * An example might help:
+         * an assembled contig that's actually a repeat expansion from 1 repeat to 2 repeats with pseudo-homology:
+         * TGCCAGGTTACATGGCAAAGAGGGTAGATATGGGGAGCTGTGAAGAATGGAGCCAGTAATTAAATTCACTGAAGTCTCCACAGGAGGGCAAGGTGGACAATCTGTCCCATAGGAGGGGGATTCATGAGGGGAGCTGTGAAGAATGGAGCCAGTAATTAAATTCACTGAAGTCTCCACAGGAGGGCAAGGTGGACAATCTGTCCCATAGGAGGGGGATTCAGGAGGGCAGCTGTGGATGGTGCAAATGCCATTTATGCTCCTCTCCACCCATATCC
+         * can be aligned to chr18:312XXX:
+         * the 1st alignment chr18:312579-718 140M135S, which can be broken into the following part
+         * 31:  TGCCAGGTTACATGGCAAAGAGGGTAGATAT
+         * 109: GGGGAGCTGTGAAGAATGGAGCCAGTAATTAAATTCACTGAAGTCTCCACAGGAGGGCAAGGTGGACAATCTGTCCCATAGGAGGGGGATTCATGAGGGGAGCTGTGAA
+         * 135: GAATGGAGCCAGTAATTAAATTCACTGAAGTCTCCACAGGAGGGCAAGGTGGACAATCTGTCCCATAGGAGGGGGATTCAGGAGGGCAGCTGTGGATGGTGCAAATGCCATTTATGCTCCTCTCCACCCATATCC
+         * And the arithmetic works this way:
+         * 31 + 109 = 140
+         * 109 = 96 + 13
+         * where 31 is the left flanking region before the repeated unit, which itself is 96 bases long (see below),
+         * the number 13 is the length of the pseudo-homology between the starting bases of the repeated sequence and the right flanking region
+         * a clearer picture emerges when we look at the 2nd alignment chr18:312610-757 127S148M, which can be broken into
+         * 31: TGCCAGGTTACATGGCAAAGAGGGTAGATAT
+         * 96: GGGGAGCTGTGAAGAATGGAGCCAGTAATTAAATTCACTGAAGTCTCCACAGGAGGGCAAGGTGGACAATCTGTCCCATAGGAGGGGGATTCATGA
+         * 96: GGGGAGCTGTGAAGAATGGAGCCAGTAATTAAATTCACTGAAGTCTCCACAGGAGGGCAAGGTGGACAATCTGTCCCATAGGAGGGGGATTCAGGA
+         * 13: GGGCAGCTGTGGA
+         * 39: TGGTGCAAATGCCATTTATGCTCCTCTCCACCCATATCC
+         * And the arithmetic works this way:
+         * 31 + 96 = 127
+         * 96 + 13 + 39 = 148
+         */
+        @VisibleForTesting
+        TandemRepeatComplication(final int distBetweenAlignRegionsOnRef, final int distBetweenAlignRegionsOnCtg) {
```

Comment:
What if you called this inner class `TandemRepeatStructure` or something? Having an inner class that ends in the same word (Complication) as the outer class made me think it would be a subclass at first reading, but this is really storing extra stuff.